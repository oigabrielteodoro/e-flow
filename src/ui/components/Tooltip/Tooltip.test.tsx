import React from 'react'
import { render, screen, fireEvent, waitFor } from '@testing-library/react'

import { Tooltip } from '.'

describe('<Tooltip />', () => {
  it('should render correctly', () => {
    const { container } = render(<Tooltip title='Example'>Hover me</Tooltip>)

    expect(screen.getByText('Example')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render tooltip on hover', async () => {
    render(<Tooltip title='Example'>Hover me</Tooltip>)

    const elementToHover = screen.getByText('Hover me')

    fireEvent.mouseEnter(elementToHover)

    await waitFor(() => expect(screen.getByText('Example')).toBeInTheDocument())
  })

  it('should render with variant chart', async () => {
    render(<Tooltip title='Example'>Hover me</Tooltip>)

    expect(screen.getByText('Example')).toBeInTheDocument()
  })
})
