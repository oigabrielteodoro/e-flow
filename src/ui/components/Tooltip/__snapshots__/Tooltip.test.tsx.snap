// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> should render correctly 1`] = `
.c0 {
  position: relative;
}

.c0:hover span {
  opacity: 1;
  visibility: visible;
}

.c0 > span {
  color: #ffffff;
  background: #0047BB;
  border-radius: 0.4rem;
  padding: 0.5rem 1.2rem;
  white-space: nowrap;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
  opacity: 0;
  visibility: hidden;
  position: absolute;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  top: calc(-100% + -1.7rem);
  z-index: 2;
}

.c0 > span::before {
  content: '';
  position: absolute;
  border-style: solid;
  border-width: 0.9rem 0.9rem 0 0.9rem;
  border-color: #0047BB transparent transparent transparent;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  bottom: -0.9rem;
}

<div
  class="c0"
>
  <span>
    Example
  </span>
  Hover me
</div>
`;
